### ISSUE TITLE
False output of 'eval' result in the training log

### ISSUE THREAD

**Issue by keke1u** [open-mmlab/mmsegmentation#694]
Hi, I found that in the latest version, the log.json file will show wrong mode when evaluating the model. And the learning rate will also decay when evaluating, if I set the "poly" training policy. Maybe validation was regarded as training step mistakenly?

I also found the problem also exists in the [vit](https://download.openmmlab.com/mmsegmentation/v0.5/vit/upernet_deit-s16_512x512_160k_ade20k/20210621_160903.log.json). model log published in the md doc.

`{"mode": "train", "epoch": 13, "iter": 15950, "lr": 5e-05, "memory": 10795, "data_time": 0.00645, "decode.loss_seg": 0.55248, "decode.acc_seg": 65.31351, "aux.loss_seg": 0.23862, "aux.acc_seg": 64.50345, "loss": 0.7911, "time": 0.27292}
{"mode": "train", "epoch": 13, "iter": 250, "lr": 5e-05, "memory": 10891, "aAcc": 0.7752, "mIoU": 0.3643, "mAcc": 0.4847, "IoU.wall": 0.7016, "IoU.building": 0.7759, "IoU.sky": 0.9207, "IoU.floor": 0.7589, "IoU.tree": 0.6992, "IoU.ceiling": 0.7915, "IoU.road": 0.805, "IoU.bed ": 0.7959, "IoU.windowpane": 0.5133, "IoU.grass": 0.6899, "IoU.cabinet": 0.5145, "IoU.sidewalk": 0.5959, "IoU.person": 0.6901, "IoU.earth": 0.3444, "IoU.door": 0.2729, "IoU.table": 0.4354,`

**Comment by Junjun2016** [comment]
Hi @keke1u 
Thanks for your feedback!
We will check and fix it ASAP!

**Comment by Junjun2016** [comment]
Hi @keke1u 
We have located the bug.
The bug was caused by MMSegmentation and MMCV.
[`runner.log_buffer.clear()`](https://github.com/open-mmlab/mmsegmentation/blob/v0.13.0/mmseg/core/evaluation/eval_hooks.py) was removed after 0.13.0 which leads to training and time log information are still in log_buffer.
The priority of `IterTimerHook` is lower than `DistEvalHook`.
We will fix both ASAP!
